ansible_collections/arista/avd/docs/getting-started/avd-aap.md

```diff
@@ -0,0 +1,403 @@
+
+You can place the Ansible builder dependencies with your current project or leverage a separate project entirely for your EE builds. Please note that this example was created with version 4.7.1 of the AVD collection. Dependencies may change between versions.
+
+```shell
+❯ tree -I venv/
+.
+├── execution-environment.yml
+├── requirements.txt
+└── requirements.yml
+```
+
+=== "execution-environment.yml"
+
+    ```yaml
+    ---
+    version: 3 #(1)
+
+    images: #(2)
+      base_image:
+        name: quay.io/centos/centos:stream8
+
+    dependencies: #(3)
+      python_interpreter:
+        package_system: python311
+        python_path: /usr/bin/python3.11
+
+      ansible_core: #(4)
+        package_pip: ansible-core<2.17.0
+
+      ansible_runner: #(5)
+        package_pip: ansible-runner
+
+      galaxy: requirements.yml #(6)
+
+      python: requirements.txt #(7)
+
+    ```
+
+    1. We are leveraging Ansible Builder 3.x, which requires version 3 of the definition file. If you leverage an older version of Ansible Builder, you may need to use `version: 1`.
+    2. You may use any base container image. Please see the official documentation for more examples.
+    3. Specifying what version of Python we would like installed during the container build.
+    4. The version of ansible-core you need will greatly depend on your specific workflows.
+    5. The `ansible-runner` package is required when building an EE.
+    6. This is a pointer to any additional collections we want installed on our container.
+    7. This is a pointer to any additional Python packages we require for our workflow.
+
+=== "requirements.yml"
+
+    ```yaml
+    ---
+    collections: #(1)
+      - name: arista.avd
+        version: 4.7.1
+
+    ```
+
+    1. Installing the `arista.avd` collection for this workflow will ensure that any other required collections are also installed.
+
+=== "requirements.txt"
+
+    ```text
+    netaddr>=0.7.19
+    Jinja2>=3.0.0
+    treelib>=1.5.5
+    cvprac>=1.3.1
+    jsonschema>=4.5.1,<4.18
+    requests>=2.27.0
+    PyYAML>=6.0.0
+    md-toc>=7.1.0
+    deepmerge>=1.1.0
+    cryptography>=38.0.4
+    # No anta requirement until the eos_validate_state integration is out of preview.
+    # anta>=1.0.0
+    aristaproto>=0.1.1
+
+    ```
+
+    The Python dependencies were taken from the [collection installation](../../docs/installation/collection-installation.md#python-requirements-installation) instructions. Please update for your specific version of the `arista.avd` collection.
+
+#### Build the image
+
+The command below will use Ansible Builder to start building our custom EE. In this case, we leverage Podman as a container runtime and tag the image appropriately.
+
+```shell
+ansible-builder build --container-runtime podman -v 3 --tag username/image-name
+```
+
+Once complete, you can push the image to a public or private container registry.
+
+```shell
+podman push IMAGEID docker://docker.io/username/image-name
+```
```

Comment:
Add some clarification that this can be any other container registry of customer choice, including internal container registry if required for security reasons. I'd also mention the fact that Docker hub can rate-limit unauthenticated pulls. That breaks things quite often.

Reply:
The rate limit may be out of scope of this guide. I will add a note and mention other registries are available like GHCR.